### ISSUE TITLE
Lots of "unused-but-set-variable" warnings

### ISSUE THREAD

**Issue by Wohlstand** [libxmp/libxmp#432]
While I polishing the build, I met a lot of "unused-but-set-variable" warnings at my side (GCC 9.3.0):
```
/home/vitaly/_git_repos/libxmp/src/loaders/coco_load.c: In function ‘coco_load’:
/home/vitaly/_git_repos/libxmp/src/loaders/coco_load.c:187:15: warning: variable ‘pat_ptr’ set but not used [-Wunused-but-set-variable]
  187 |  int seq_ptr, pat_ptr, smp_ptr[100];
      |               ^~~~~~~
/home/vitaly/_git_repos/libxmp/src/loaders/gal4_load.c: In function ‘get_inst’:
/home/vitaly/_git_repos/libxmp/src/loaders/gal4_load.c:221:31: warning: variable ‘fade’ set but not used [-Wunused-but-set-variable]
  221 |  int val, vwf, vra, vde, vsw, fade;
      |                               ^~~~
/home/vitaly/_git_repos/libxmp/src/loaders/med3_load.c: In function ‘med3_load’:
/home/vitaly/_git_repos/libxmp/src/loaders/med3_load.c:352:12: warning: variable ‘tracks’ set but not used [-Wunused-but-set-variable]
  352 |   uint8 b, tracks;
      |            ^~~~~~
/home/vitaly/_git_repos/libxmp/src/loaders/med4_load.c: In function ‘med4_load’:
/home/vitaly/_git_repos/libxmp/src/loaders/med4_load.c:807:23: warning: variable ‘ver’ set but not used [-Wunused-but-set-variable]
  807 |   int32 id, size, s2, ver;
      |                       ^~~
/home/vitaly/_git_repos/libxmp/src/loaders/med4_load.c:180:13: warning: variable ‘hexvol’ set but not used [-Wunused-but-set-variable]
  180 |  int flags, hexvol = 0;
      |             ^~~~~~
/home/vitaly/_git_repos/libxmp/src/loaders/mgt_load.c: In function ‘mgt_load’:
/home/vitaly/_git_repos/libxmp/src/loaders/mgt_load.c:218:16: warning: variable ‘f2p’ set but not used [-Wunused-but-set-variable]
  218 |    uint8 note, f2p;
      |                ^~~
/home/vitaly/_git_repos/libxmp/src/loaders/mgt_load.c:64:51: warning: variable ‘smp_ptr’ set but not used [-Wunused-but-set-variable]
   64 |  int sng_ptr, seq_ptr, ins_ptr, pat_ptr, trk_ptr, smp_ptr;
      |                                                   ^~~~~~~
/home/vitaly/_git_repos/libxmp/src/loaders/mmd_common.c: In function ‘mmd_load_hybrid_instrument’:
/home/vitaly/_git_repos/libxmp/src/loaders/mmd_common.c:414:14: warning: variable ‘type’ set but not used [-Wunused-but-set-variable]
  414 |  int length, type;
      |              ^~~~
/home/vitaly/_git_repos/libxmp/src/loaders/st_load.c: In function ‘st_load’:
/home/vitaly/_git_repos/libxmp/src/loaders/st_load.c:221:7: warning: variable ‘size’ set but not used [-Wunused-but-set-variable]
  221 |  long size;
      |       ^~~~
/home/vitaly/_git_repos/libxmp/src/loaders/sym_load.c: In function ‘sym_load’:
/home/vitaly/_git_repos/libxmp/src/loaders/sym_load.c:256:15: warning: variable ‘infolen’ set but not used [-Wunused-but-set-variable]
  256 |  int /*ver,*/ infolen, sn[64];
      |               ^~~~~~~
/home/vitaly/_git_repos/libxmp/src/depackers/inflate.c: In function ‘decompress’:
/home/vitaly/_git_repos/libxmp/src/depackers/inflate.c:717:26: warning: variable ‘curr_huffman_leaf’ set but not used [-Wunused-but-set-variable]
  717 |   struct huffman_tree_t *curr_huffman_leaf;
      |                          ^~~~~~~~~~~~~~~~~
/home/vitaly/_git_repos/libxmp/src/depackers/inflate.c: In function ‘libxmp_inflate’:
/home/vitaly/_git_repos/libxmp/src/depackers/inflate.c:1025:16: warning: variable ‘DICT’ set but not used [-Wunused-but-set-variable]
 1025 |   unsigned int DICT;
      |                ^~~~
/home/vitaly/_git_repos/libxmp/src/loaders/prowizard/p40.c: In function ‘depack_p4x’:
/home/vitaly/_git_repos/libxmp/src/loaders/prowizard/p40.c:61:13: warning: variable ‘npat’ set but not used [-Wunused-but-set-variable]
   61 |  uint8 len, npat, nsmp;
      |             ^~~~
/home/vitaly/_git_repos/libxmp/src/loaders/prowizard/p61a.c: In function ‘depack_p61a’:
/home/vitaly/_git_repos/libxmp/src/loaders/prowizard/p61a.c:40:9: warning: variable ‘Unpacked_Sample_Data_Size’ set but not used [-Wunused-but-set-variable]
   40 |     int Unpacked_Sample_Data_Size;
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~
/home/vitaly/_git_repos/libxmp/src/loaders/prowizard/p61a.c:30:11: warning: variable ‘PACK’ set but not used [-Wunused-but-set-variable]
   30 |     uint8 PACK[31];
      |           ^~~~
[ 50%] Built target xmp_shared
/home/vitaly/_git_repos/libxmp/src/loaders/coco_load.c: In function ‘coco_load’:
/home/vitaly/_git_repos/libxmp/src/loaders/coco_load.c:187:15: warning: variable ‘pat_ptr’ set but not used [-Wunused-but-set-variable]
  187 |  int seq_ptr, pat_ptr, smp_ptr[100];
      |               ^~~~~~~
/home/vitaly/_git_repos/libxmp/src/loaders/gal4_load.c: In function ‘get_inst’:
/home/vitaly/_git_repos/libxmp/src/loaders/gal4_load.c:221:31: warning: variable ‘fade’ set but not used [-Wunused-but-set-variable]
  221 |  int val, vwf, vra, vde, vsw, fade;
      |                               ^~~~
/home/vitaly/_git_repos/libxmp/src/loaders/med3_load.c: In function ‘med3_load’:
/home/vitaly/_git_repos/libxmp/src/loaders/med3_load.c:352:12: warning: variable ‘tracks’ set but not used [-Wunused-but-set-variable]
  352 |   uint8 b, tracks;
      |            ^~~~~~
/home/vitaly/_git_repos/libxmp/src/loaders/med4_load.c: In function ‘med4_load’:
/home/vitaly/_git_repos/libxmp/src/loaders/med4_load.c:807:23: warning: variable ‘ver’ set but not used [-Wunused-but-set-variable]
  807 |   int32 id, size, s2, ver;
      |                       ^~~
/home/vitaly/_git_repos/libxmp/src/loaders/med4_load.c:180:13: warning: variable ‘hexvol’ set but not used [-Wunused-but-set-variable]
  180 |  int flags, hexvol = 0;
      |             ^~~~~~
/home/vitaly/_git_repos/libxmp/src/loaders/mgt_load.c: In function ‘mgt_load’:
/home/vitaly/_git_repos/libxmp/src/loaders/mgt_load.c:218:16: warning: variable ‘f2p’ set but not used [-Wunused-but-set-variable]
  218 |    uint8 note, f2p;
      |                ^~~
/home/vitaly/_git_repos/libxmp/src/loaders/mgt_load.c:64:51: warning: variable ‘smp_ptr’ set but not used [-Wunused-but-set-variable]
   64 |  int sng_ptr, seq_ptr, ins_ptr, pat_ptr, trk_ptr, smp_ptr;
      |                                                   ^~~~~~~
/home/vitaly/_git_repos/libxmp/src/loaders/mmd_common.c: In function ‘mmd_load_hybrid_instrument’:
/home/vitaly/_git_repos/libxmp/src/loaders/mmd_common.c:414:14: warning: variable ‘type’ set but not used [-Wunused-but-set-variable]
  414 |  int length, type;
      |              ^~~~
/home/vitaly/_git_repos/libxmp/src/loaders/st_load.c: In function ‘st_load’:
/home/vitaly/_git_repos/libxmp/src/loaders/st_load.c:221:7: warning: variable ‘size’ set but not used [-Wunused-but-set-variable]
  221 |  long size;
      |       ^~~~
/home/vitaly/_git_repos/libxmp/src/loaders/sym_load.c: In function ‘sym_load’:
/home/vitaly/_git_repos/libxmp/src/loaders/sym_load.c:256:15: warning: variable ‘infolen’ set but not used [-Wunused-but-set-variable]
  256 |  int /*ver,*/ infolen, sn[64];
      |               ^~~~~~~
/home/vitaly/_git_repos/libxmp/src/depackers/inflate.c: In function ‘decompress’:
/home/vitaly/_git_repos/libxmp/src/depackers/inflate.c:717:26: warning: variable ‘curr_huffman_leaf’ set but not used [-Wunused-but-set-variable]
  717 |   struct huffman_tree_t *curr_huffman_leaf;
      |                          ^~~~~~~~~~~~~~~~~
/home/vitaly/_git_repos/libxmp/src/depackers/inflate.c: In function ‘libxmp_inflate’:
/home/vitaly/_git_repos/libxmp/src/depackers/inflate.c:1025:16: warning: variable ‘DICT’ set but not used [-Wunused-but-set-variable]
 1025 |   unsigned int DICT;
      |                ^~~~
/home/vitaly/_git_repos/libxmp/src/loaders/prowizard/p40.c: In function ‘depack_p4x’:
/home/vitaly/_git_repos/libxmp/src/loaders/prowizard/p40.c:61:13: warning: variable ‘npat’ set but not used [-Wunused-but-set-variable]
   61 |  uint8 len, npat, nsmp;
      |             ^~~~
/home/vitaly/_git_repos/libxmp/src/loaders/prowizard/p61a.c: In function ‘depack_p61a’:
/home/vitaly/_git_repos/libxmp/src/loaders/prowizard/p61a.c:40:9: warning: variable ‘Unpacked_Sample_Data_Size’ set but not used [-Wunused-but-set-variable]
   40 |     int Unpacked_Sample_Data_Size;
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~
/home/vitaly/_git_repos/libxmp/src/loaders/prowizard/p61a.c:30:11: warning: variable ‘PACK’ set but not used [-Wunused-but-set-variable]
   30 |     uint8 PACK[31];
      |           ^~~~
```

**Comment by AliceLR** [comment]
The autotools build system intentionally builds with `-Wno-unused-but-set-variable`, presumably because there are a lot of these and they don't hurt much.

**Comment by Wohlstand** [comment]
That right, anyway, @sezero  had told me to remove the warning checks, but later, I added them back as they were presented at the `configure.ac`.

**Comment by AliceLR** [comment]
Sorry, I didn't mean to imply this should be closed. I started looking into these since I encountered some `-Wunused-but-set-variable` warnings in a Centos 7 build for #456 (likely this was `-Wall` in GCC 4.8.5 and later got pushed to `-Wextra`, but this is a guess).

Categorization of these as I've checked them so far:
* Coconizer: this is a serious bug that prevents the correct loading of patterns.
* inflate: ignore for now since it's in the process of getting replaced.
* The Player 4.0x and 6.1a: actually unused fields, safe to comment out.

Progress here: https://github.com/libxmp/libxmp/compare/master...AliceLR:fix-unused-but-set

**Comment by AliceLR** [comment]
The MED4 issues were:

* The loader was ignoring the hex volume flag, and embarrassingly, trying to load it from the wrong byte in the first place. Both of these have been fixed. The existing MED4 test module actually uses hex volume (oops!), so I added a decimal volume test module.
* The version field when loading the `MEDV` IFF chunk is only used with `-DDEBUG` because it is loaded earlier in the loader. Making storing its major and minor version bytes to the appropriate fields seems to suppress the warning for now.

Checking into the `MEDV` warning also revealed a couple of other problems:
* The IFF section handling (aside from the early `MEDV` version check) was completely broken because of 2 bytes being read prior to reading the first IFF tag. It's not clear why, because I tested xmp against every MED4 module I have and a) none of them have this and b) IFF magically works again when I remove it.
* The annotation text wasn't being loaded to `m->comment`, which is likely a fairly common problem in libxmp loaders. This is done now, meaning `xmp -C test-dev/data/m/Synth-a-sysmic.med` now prints the title and author of that module.
